Source/Modules/scilab.cxx

```diff
@@ -1136,18 +1095,67 @@ class SCILAB:public Language {
    * ----------------------------------------------------------------------- */
 
   void saveLoaderFile(String *gatewayLibraryName) {
-    Printf(loaderScript, "];\n");
-    Printf(loaderScript, "addinter(fullfile(%s_path, '%s' + getdynlibext()), '%s', list_functions);\n",
-	   gatewayLibraryName, gatewayLibraryName, gatewayLibraryName);
-    Printf(loaderScript, "clear %s_path;\n", gatewayLibraryName);
-    Printf(loaderScript, "clear bOK;\n");
-    Printf(loaderScript, "clear ilib;\n");
-    Printf(loaderScript, "clear list_functions;\n");
+    Printf(loaderScript5, "    ];\n");
+    Printf(loaderScript6, "    ];\n");
+
+    Printf(loaderScript, "  ver = getversion('scilab');\n");
+    Printf(loaderScript, "  if ver(1) <= 5 & ver(2) <= 5 & ver(3) <= 2 then\n");
+    Printf(loaderScript, "    // version is less or equal to 5.5.2\n");
+    Printf(loaderScript, "    \n");
+    Append(loaderScript, loaderScript5);
+    Delete(loaderScript5);
+    Printf(loaderScript, "    \n");
+    Printf(loaderScript, "  else\n");
+    Printf(loaderScript, "    // version is 6.0.0 or more\n");
+    Printf(loaderScript, "    \n");
+    Append(loaderScript, loaderScript6);
+    Delete(loaderScript6);
+    Printf(loaderScript, "    \n");
+    Printf(loaderScript, "  end\n");
+
+    Printf(loaderScript, "  addinter(%s_path + '%s' + getdynlibext(), '%s', list_functions);\n", gatewayLibraryName, gatewayLibraryName, gatewayLibraryName);
+    Printf(loaderScript, "endfunction\n");
+    Printf(loaderScript, "loader_function();\n");
+    Printf(loaderScript, "clear loader_function;\n");
     Printv(loaderFile, loaderScript, NIL);
 
+    Delete(loaderScript);
     Delete(loaderFile);
   }
 
+  /* -----------------------------------------------------------------------
+   * createSmallIdentifierName()
+   * Create a Scilab small identifier to be used by Scilab 5
+   * ----------------------------------------------------------------------- */
+
+  String* createSmallIdentifierName(String* name, int outputLen = SCILAB_IDENTIFIER_NAME_CHAR_MAX) {
+    char* s;
+    if (DohCheck(name)) {
+      s = (char *) Data(name);
+    } else {
+      s = (char *) name;
+    }
```

Comment:
The implementation of `Data(obj)` includes `DohCheck(obj)` and returns `(void *) obj` for you if that returns false.  There's also a macro `Char(X)` which is defined as `((char *) Data(X))` so you can replace the code above with just:

```
char * s = Char(name);
```

Reply:
thanks for the information, I will use `Char(X)` then